web/controller/alert.ex

```diff
@@ -21,11 +21,40 @@ defmodule IotIntern.Controller.Alert do
   use Antikythera.Controller
 
   alias Antikythera.Conn
-  # 必要に応じて適宜エイリアスのコメントアウトを解除してください
-  # alias IotIntern.Error
-  # alias IotIntern.Linkit
+  alias IotIntern.Error
+  alias IotIntern.Linkit
 
-  def post_alert(%{request: %{body: _body}} = conn) do
-    Conn.json(conn, 200, %{})
+  @alert_messages %{
+    "dead_battery" => "バッテリー不足",
+    "derailment"   => "脱輪",
+    "jamming"      => "異物混入"
+  }
+
+  defmodule RequestBody do
+    use Croma
+
+    defmodule Message do
+      use Croma.SubtypeOfAtom, values: [:dead_battery, :derailment, :jamming]
+    end
+
+    use Croma.Struct, recursive_new?: true, fields: [
+      type: Message,
+    ]
+  end
+
+  def post_alert(%{request: %{body: body}} = conn) do
+    case RequestBody.new(body) do
+      {:ok, _} ->
+        message = Map.get(@alert_messages, body["type"])
```

Comment:
task5
Croma を使う場合も下記のように validate されたものを使うのが良いと思います。 

```suggestion
      {:ok, validated} ->
        message = Map.get(@alert_messages, validated.type)
```

Reply:
ありがとうございます。
下記で修正しました。

32333a8

croma と validate_request_body 関数を使う場合で、@alert_messages のキーが string と atom で異なっているのが、少し気になりますが、良い解決策を思いつかなかったためこうしています。